rmw_connext_cpp/src/rmw_take.cpp

```diff
@@ -283,7 +283,7 @@ _take_sequence(
   DDS::ReturnCode_t status = data_reader->take(
     dds_messages,
     sample_infos,
-    count,
+    static_cast<DDS_Long>(count),
```

Comment:
So, going through RTI headers a bit, it looks to me like:

```
typedef RTICdrLong DDS_Long;
```

and

```
typedef RTI_INT32 RTICdrLong;
```

and

```
typedef signed int  RTI_INT32;
```

On all of our target platforms, a signed integer is 32-bits.  This means that on 64-bit platforms theoretically `count` (which is a size_t) could hold a number much larger than what will fit into the signed 32-bit integer.  Thus, if we are going to do this, I think we need to have a check much earlier on that count <= DDS_Long::max.  Thoughts?

Reply:
If all of the supported RMW implementations take something 32 bits (either signed or unsigned), does it even make sense to use a `size_t` here? 

 I suppose I mentally defaulted to that to represent the size of something.

Reply:
Yeah, I'm not sure.  I haven't done a comparison with the other RMW's to see what they take.

Given that we are past API/feature freeze for Foxy, though, I'd say leave the API as-is and add in a check at the beginning of the method.

Reply:
Good point, done in https://github.com/ros2/rmw_cyclonedds/pull/171/commits/fe8ed460edff790cf2551177e098230bb4aadf12.